base/intfuncs.jl

```diff
@@ -174,8 +173,8 @@ function power_by_squaring(x_, p::Integer)
     elseif p == 2
         return x*x
     elseif p < 0
-        x == 1 && return copy(x)
-        x == -1 && return iseven(p) ? one(x) : copy(x)
+        isone(x) && return copy(x)
+        isone(-x) && return iseven(p) ? one(x) : copy(x)
         throw_domerr_powbysq(p)
```

Comment:
I had the same thought. I was a bit reluctant to make the error handling more complicated, since this `A^p` for p<0 seems to be a are thing for matrices. Where should `throw_domerr_powbysq(x::AbstractMatrix,p)` go? I guess dense.jl would be the right place, since intfucs.jl doesn't know anything about matrices.